src/mbgl/gfx/command_encoder.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+namespace mbgl {
+namespace gfx {
+
+class CommandEncoder {
+protected:
+    CommandEncoder() = default;
+
+public:
+    virtual ~CommandEncoder() = default;
+    CommandEncoder(CommandEncoder&&) = delete;
+    CommandEncoder(const CommandEncoder&) = delete;
+    CommandEncoder& operator=(CommandEncoder&&) = delete;
```

Comment:
not needed, as copy constructor is user-defined  (deleted), so move constructor will not be generated.